### DISCUSSION TITLE
Tank level sensor

### DISCUSSION THREAD

**Discussion by mariusl** [arachnetech/homebridge-mqttthing#551] [Ideas]
Hi,
I am trying to do a tank level sensor which is not provided for in the plugin. Is there a way that I could do that with one of the standard items.
Ideally I would like to be able set the parameters of the tank (diameter and height) to be able to determine the volume.
Any ideas please?

**Comment by HolgerAusB** [comment]
As I read it, Homekit knows a characteristic 'WaterLevel' but it is only in percent. In mqttthing I have not found this characteristic. But you can use the brightness of a light bulb or humidity, also only as a percentage. But this means that you have to calculate the percentages from the water level yourself, before you send it to the mqtt-broker.

I am not a pro, just my 2 cents

**Reply by mariusl** [reply]
I am using that with homebridge-mqtt currently but the items show up as humidity in the home shortcuts and other places. So even if you name it "Tank Level" it shows as the room has a humidity of xx% in the homes screen. Not really helpful.

**Reply by HolgerAusB** [reply]
You could use a own room named 'xxx tank level' and only that humidity sensor is in.
